charts/opentelemetry-collector/templates/prometheusrule.yaml

```
@@ -0,0 +1,14 @@
+{{- if .Values.prometheusRule.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ include "opentelemetry-collector.fullname" . }}
+  labels:
+    {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+    {{- range $key, $value := .Values.prometheusRule.extraLabels }}
+    {{- printf "%s: %s" $key (tpl $value $ | quote) | nindent 4 }}
```

is `tpl` needed? Will this work instead?
```suggestion
    {{- printf "%s: %s" $key (quote $value) | nindent 4 }}
```

---

If the suggestion works, then [servicemonitor.yaml](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/servicemonitor.yaml) and [podmonitor.yaml](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/podmonitor.yaml) should be updated to match.

---

Although I think changing servicemonitor.yaml and podmonitor.yaml would be a breaking change.  Isn't that tpl function letting user add things like `{{ .Values.fullnameOverride }}` in the `.extraLabels` section?

---

Oh ok, we can keep his for consistency them. probably can be addressed separately.

> Isn't that tpl function letting user add things like {{ .Values.fullnameOverride }} in the .extraLabels section?

I don't think so. I don't really see a reason to have `tpl` here